### DISCUSSION TITLE
Composite docker action

### DISCUSSION THREAD

**Discussion by fortierq** [community/community#25623] [Actions]
<p>I would like to write a generic action combining 3 steps, one of which requires to be run with a custom Docker image.</p>
<p>It looks like this:</p>
<pre><code class="lang-auto">runs:
  using: "composite"
  steps:
    - uses: actions/checkout@v2
    - uses: jitterbit/get-changed-files@v1
    - run: |
        # commands that need to be run in a container
</code></pre>
<p>Can I add something like this below (which is what I use in my standalone action) ?</p>
<pre><code class="lang-auto">jobs:
  myjob:
    runs-on: ubuntu-latest
    container:
      image: # my image
    steps:
      - ...
</code></pre>
<p>Or should I use a docker action? (but I don’t think I can use actions/checkout@v2 and jitterbit/get-changed-files in it, which is unfortunate)</p>

**Comment by ChristopherHX** [comment, marked as answer]
<blockquote>
<p>Can I add something like this below (which is what I use in my standalone action) ?</p>
</blockquote>
<p>No as far as I know this exists only on Job level.</p>
<p>However the runner mounts your workspace (with all changes of your previous steps) into the docker action so you should be able to to replace run with <code>uses: docker://</code> without a custom docker action.</p>
<pre><code class="lang-yaml">runs:
  using: "composite"
  steps:
    - uses: actions/checkout@v2
    - uses: jitterbit/get-changed-files@v1
    - name: Explain why this job ran
      uses: docker://# my image
      with:
        entrypoint:  # command executable that need to be run in a container
        args:  # args of the command that need to be run in a container
</code></pre>
<p>See <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepswithargs" class="inline-onebox" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a>, <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-a-docker-hub-action" class="inline-onebox" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a>.</p>

**Reply by kyrylogy** [reply]
Has this been updated? I saw https://github.com/appleboy/ssh-action and it looks like they're using it in some new way that I also need but can't understand how it works.
